massa-api/src/public.rs

```diff
@@ -413,41 +413,16 @@ impl Endpoints for API<Public> {
         let cfg = self.0.consensus_config.clone();
         let api_cfg = self.0.api_settings;
         let pool_command_sender = self.0.pool_command_sender.clone();
+        let execution_controller = self.0.execution_controller.clone();
         let compensation_millis = self.0.compensation_millis;
 
-        // todo make better use of SCE ledger info
-
-        // map SCE ledger info and check for address length
-        let sce_ledger_info = if addresses.len() as u64 > api_cfg.max_arguments {
-            Err(ApiError::TooManyArguments("too many arguments".into()))
-        } else {
-            // get SCE ledger info
-            let mut sce_ledger_info: Map<Address, SCELedgerInfo> =
-                Map::with_capacity_and_hasher(addresses.len(), BuildMap::default());
-            for addr in &addresses {
-                let active_entry = match self
-                    .0
-                    .execution_controller
-                    .get_final_and_active_ledger_entry(addr)
-                    .1
-                {
-                    None => continue,
-                    Some(v) => SCELedgerInfo {
-                        balance: v.parallel_balance,
-                        module: Some(v.bytecode),
-                        datastore: v.datastore.into_iter().collect(),
-                    },
-                };
-                sce_ledger_info.insert(*addr, active_entry);
-            }
-            Ok(sce_ledger_info)
-        };
-
         let closure = async move || {
```

Comment:
A good follow-up issue would be to really split this enormous closure into several chunks of code ...

Reply:
#2399